domain/urgentWishForChildren.unit.ts

```diff
@@ -0,0 +1,90 @@
+import { buildBaseObjects, expectMutationsOfType, shouldPlaceDwarf } from "./testUtils";
+import { UrgentWishForChildren } from "./urgentWishForChildren";
+import { Dwarf, Player } from "./entity/Player";
+import { ActionSpace } from "./entity/ActionSpace";
+import { EntityMutation, isMutationOfType } from "./entity/Mutation";
+import { expect } from "chai";
+
+describe("Urgent Wish for Children", () => {
+    shouldPlaceDwarf(UrgentWishForChildren.execute);
+
+    describe("should create and place a new dwarf", () => {
+        it("should add new dwarf to player", function () {
+            const { game, player } = buildBaseObjects();
+            const playerInitialDwarfsNb = player.dwarfs.size;
+
+            const mutations = UrgentWishForChildren.execute(game, player.id);
+
+            expectMutationsOfType(mutations, Player).toVerifyOnce(
+                (mutation) => mutation.diff.dwarfs?.size === playerInitialDwarfsNb + 1
+            );
+        });
+
+        it("should add new dwarf to action space", function () {
+            const { game, player } = buildBaseObjects();
+
+            const mutations = UrgentWishForChildren.execute(game, player.id);
+
+            expectMutationsOfType(mutations, ActionSpace).toVerifyOnce(
+                (mutation) => !!mutation.diff.newBornDwarf
+            );
+        });
+
+        it("new dwarf should be the same for player and action space", function () {
+            const { game, player } = buildBaseObjects();
+
+            const mutations = UrgentWishForChildren.execute(game, player.id);
+
+            const actionSpaceMutation = mutations
```

Comment:
Detail: I would have created 2 sub-functions here `getActionSpaceDwarf()` and `getPlayerDwarf()` inside the `it`

(but feel free to keep it like that, it doesn't matter much)

Reply:
Agree, it makes the test more understandable. changed!